File: CS50x/wk2/problems/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

bool readingLevel(int length, string text);

int main(void)
{
    string text = get_string("Text: \n");
    int length = strlen(text);
    readingLevel(length, text);
}

bool readingLevel(int length, string text)
{
    int words = 0;
    int letters = 0;
    int sentences = 0;
    for (int i = 0; i < length; i++)
    {
        char c = text[i];
        if (isalpha(c))
        {
            letters++;
        }
        else if (isspace(c))
        {
            words++;
        }
        else if (c == '.' || c == '!' || c == '?')
        {
            sentences++;
        }
    }
    words++;
    float L = letters / ((float) words / 100);
    float S = sentences / ((float) words / 100);
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    if (index < 1)
    {
        printf("Before Grade 1\n");
        return false;
    }
    else if (index > 15)
    {
        printf("Grade 16+\n");
        return false;
    }
    else
    {
        printf("Grade %i\n", index);
        return true;
    }
}

File: CS50x/wk6/problems/sentimental-readability/readability.py
# TODO


def main():
    input = get_user_input()
    print(grade_level(input))


def get_user_input():
    s = input("Text: ")
    return s


def grade_level(input):
    level = round(coleman_liau_index(input))
    if level >= 16:
        return "Grade 16+"
    elif level < 1:
        return "Before Grade 1"
    else:
        return f"Grade {level}"


def coleman_liau_index(input):
    # Step 1: Count total number of letters
    letters = sum(1 for char in input if char.isalpha())

    # Step 2: Count total number of sentences
    modified_input = input.replace("...", "###")
    periods = modified_input.count(".")
    exclamations = modified_input.count("!")
    question = modified_input.count("?")
    ellipses = modified_input.count("###")
    sentences = periods + exclamations + question + ellipses

    # Step 3: Count total number of words
    words = len(input.split())

    # Step 4: Calculate average number of letters per word
    letters_per_word = letters / words

    # Step 5: Calculate average number of sentences per word
    sentences_per_word = sentences / words

    # Step 6: Calculate average number of letters per 100 words
    L = letters_per_word * 100

    # Step 7: Calculate average number of sentences per 100 words
    S = sentences_per_word * 100

    # Step 8: Calculate the Coleman-Liau index
    return 0.0588 * L - 0.296 * S - 15.8


main()
